consumers.go

```diff
@@ -0,0 +1,58 @@
+package rabbithole
+
+import (
+	"net/url"
+)
+
+type BriefQueueInfo struct {
+	Name  string `json:"name"`
+	Vhost string `json:"vhost"`
+}
+
+type BriefChannelDetail struct {
+	ConnectionName string `json:"connection_name"`
+	Name           string `json:"name"`
+	Node           string `json:"node"`
+	Number         int    `json:"number"`
+	PeerHost       string `json:"peer_host"`
+	PeerPort       int    `json:"peer_port"`
+	User           string `json:"user"`
+}
+
+type ConsumerInfo struct {
+	Arguments      map[string]interface{} `json:"arguments"`
+	AckRequired    bool                   `json:"ack_required"`
+	ChannelDetails BriefChannelDetail     `json:"channel_details"`
+	ConsumerTag    string                 `json:"consumer_tag"`
+	Exclusive      bool                   `json:"exclusive"`
+	PrefetchCount  int                    `json:"prefetch_count"`
+	Queue          BriefQueueInfo         `json:"queue"`
+}
+
+// ListConsumers lists all consumers in the cluster.
+func (c *Client) ListConsumers() (rec []ConsumerInfo, err error) {
+	req, err := newGETRequest(c, "consumers")
+	if err != nil {
+		return []ConsumerInfo{}, err
```

Comment:
You should be able to `return nil, err` which should be preferred as you wouldn't be allocating when returning an error.
Since you've named your return values you should even be able to just `return`.

Reply:
I addressed those issues in `13d99fc`.